lib/stat_tracker.rb

```diff
@@ -146,13 +146,32 @@ def most_accurate_team
   def least_accurate_team
 
   end
+  
 
-  def most_tackles
-
+  def most_tackles(season_id)
+    games_in_season = @game_teams.find_all do |game|
+      game.game_id[0..3] == season_id[0..3]
+    end
+    games_group_by_team = games_in_season.group_by { |team| team.team_id }
+    team_tackles = games_group_by_team.map do |team, tackles|
+    [team, tackles.map { |game| game.tackles}]
+    end
+    tackle_totals = team_tackles.map {|team, tackles| [team, tackles.sum]}.to_h
+    highest_total = tackle_totals.max_by { |team, tackles| tackles}
+    @teams.find { |team| team.team_id == highest_total.first}.team_name
   end
 
-  def fewest_tackles
-
+  def fewest_tackles(season_id)
+    games_in_season = @game_teams.find_all do |game|
+      game.game_id[0..3] == season_id[0..3]
+    end
+    games_group_by_team = games_in_season.group_by { |team| team.team_id }
+    team_tackles = games_group_by_team.map do |team, tackles|
+    [team, tackles.map { |game| game.tackles}]
+    end
```

Comment:
we might be able to break some of this out into helper methods that would have use in other methods for sure!